### DISCUSSION TITLE
Help - Establish communications with GROWATT SPF

### DISCUSSION THREAD

**Discussion by marbeja** [wills106/homeassistant-solax-modbus#761] [Q&A]
### Describe the bug

I try to connect my Nuc based HA via this Solax modbus but can't get anything out of it.
The serial to Ethernet converter blinks Rx and tx led, IP address correct, tried Port 23 25 26 502 506 but never got anything, tried TCP server modbus  also with RTU and without, tried modbus address 1 2 3 4 30001 30.000
always the same, no data. Will try to upload the log files here. 
I also tried two USB to Rs485 sticks no data, but the wires are having the typical 0.5V from Rs485 I worked before (orange pair)
Logfile says unable to read serial number %9

### Integration Version

2024 02 05

### Homeassistant core version

114 c01 05

### Inverter brand

GROWATT spf5000es

### Plugin used

plugin_growatt.py

### Serial prefix

JZM2FAD

### Connection Method

Use iot just tcp232-410S

### Detailed Error Log

Will add from pc

**Comment by marbeja** [comment]
Core 2024.2.1
Supervisor 2024.01.1
Operating System 11.5
Frontend 20240207.1

**Comment by marbeja** [comment]
[home-assistant_solax_modbus_2024-02-20T09-54-30.592Z.log](https://github.com/wills106/homeassistant-solax-modbus/files/14342583/home-assistant_solax_modbus_2024-02-20T09-54-30.592Z.log)

**Comment by marbeja** [comment]
sorry as newbie I cannot edit the issue with adding the error log, it is too big!

**Comment by marbeja** [comment]
the inverter and  is set to 9600bd corretctly, no idea where these 19200 baud come from!
Sorry my logfile is 4,5 MB but there is no possibility to delete it in HA under logs

**Comment by marbeja** [comment]
what does SPF -WIP mean ?  I know that there are several SPF Inverters but never heard of  WIP, neither google :(  

Growatt:

    AC Battery Storage:
        SPA

    Hybrid:
        SPF - WIP
        SPH
        TL-XH (MIN & MOD)

    PV Only:
        MAC
        MAX
        MID
        TL-X

**Comment by wills106** [comment]
WIP = Work In Progress You can follow the progress #686 

As you are using an RS485-USB Adaptor you need to select `Serial` on the first window (1st Image) and then it will ask you for the `Serial port name` on the next page (3rd Image)
https://homeassistant-solax-modbus.readthedocs.io/en/latest/installation/#pair-your-inverter

**Comment by marbeja** [comment]
thanks for the speedy reply, I tried **both, the USB and the  RS485 to Ethernet**  but both don't work!
I used lsusb -v in console to verify my 2 USB / 485 converters and it seems the are not recognized, so I will concentrate on the additional ethernet connection:

![grafik](https://github.com/wills106/homeassistant-solax-modbus/assets/13425998/11e11cb2-9c71-4b3b-ad16-586b2e705d13)

**Comment by marbeja** [comment]
![grafik](https://github.com/wills106/homeassistant-solax-modbus/assets/13425998/d4eb235e-8902-4f7a-b130-0bd14befa9b2)

is that correct  to put  port 502 everywhere?
![grafik](https://github.com/wills106/homeassistant-solax-modbus/assets/13425998/4d4d7a40-7e13-43f0-8412-60454d54c2c9)

**Comment by marbeja** [comment]
![grafik](https://github.com/wills106/homeassistant-solax-modbus/assets/13425998/7b8fbacc-c820-48ca-95af-0b79fca9caa3)

I tried modbus adress 1 to 3 and without parallel even I have 2 inverters 
![grafik](https://github.com/wills106/homeassistant-solax-modbus/assets/13425998/ade92bb5-faef-4391-a286-fc68fdeb7765)

do I have to mark `Modbus  RTU over  TOP here for the rather old SPF ??

**Comment by wills106** [comment]
EPS, Dry Contact and Parallel Mode wont do anything on your Inverter.

Can't you see the modbus address on the Inverter LCD?

I would put Misc Config back to default. Or factory reset the URS.

Also it looks like you are pointing the Integration at 192.168.1.7 but you are configuring the URS on 192.168.1.10 though.

Also are you connecting A to A and B to B between the URS and Inverter? They don't go A-B and B-A like on a RS232 setup.

**Reply by wills106** [reply]
I have moved this to a discussion as it isn't an Integration issue.

**Comment by abspoel99** [comment]
Here is my setup, hope it helps.

Inverter: Growatt 10KTL3-XH
Modbus: waveshare eth to modbus 
Intergration: solax inverter modbus

![IMG_7365](https://github.com/wills106/homeassistant-solax-modbus/assets/29400615/d0d531b6-6ee5-4b07-a96c-bc5179526499)
![IMG_7364](https://github.com/wills106/homeassistant-solax-modbus/assets/29400615/2c55a81e-0c7b-4490-b834-f4dd1d600351)
![IMG_7363](https://github.com/wills106/homeassistant-solax-modbus/assets/29400615/eec73ccb-7537-4ef8-8972-956169db69fe)
